src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -266,7 +277,9 @@ class SetupPy(Goal):
     pex_path="./setuptools.pex",
     pex_args=('setup.py', *req.args),
     input_files=merged_input_files,
-    output_files=('**',),
+    # setuptools commands that create dists write them to dist/.
+    # TODO: Could there be other useful files to capture?
+    output_directories=('dist/',),
```

Comment:
This is only the default. The `sdist` and `bdist_wheel` commands both accept a `--dist-dir` argument. Probably worth a TODO + issue to follow up and address later.

Reply:
Added this to https://github.com/pantsbuild/pants/issues/8912